main.py

```diff
@@ -46,8 +47,12 @@ def make_graph(percent: float, blocks: str) -> str:
 
 def get_stats() -> str:
     '''Gets API data and returns markdown progress'''
+    encoded_key: str = str(base64.b64encode(waka_key.encode('utf-8')), 'utf-8')
     data = requests.get(
-        f"https://wakatime.com/api/v1/users/current/stats/last_7_days?api_key={waka_key}").json()
+        f"{api_base_url.rstrip('/')}/v1/users/current/stats/last_7_days",
+        headers={
+            "Authorization": f"Basic {encoded_key}"
+        }).json()
```

Comment:
I can't believe I wrote like this. This was an hour's hack when the Profile Readme feature was introduced. Thanks for the rewrite ✨